bika/lims/tests/doctests/API_analysis.rst

```diff
@@ -500,3 +501,163 @@ Now, check for out-of-range results:
+    >>> analyses.sort(key=lambda analysis: analysis.getKeyword(), reverse=False)
+
+Set results for analysis `Au` (min: -5, max: 5, warn_min: -5.5, warn_max: 5.5):
+
+    >>> au_analysis = analyses[0]
+    >>> au_analysis.setResult(-5)
+    >>> is_out_of_range(au_analysis)
+    (True, False)
+
+    >>> au_analysis.setResult(5)
+    >>> is_out_of_range(au_analysis)
+    (True, False)
+
+
+Check if results are out of range when left-open interval is used
+-----------------------------------------------------------------
+
+Set left-open interval for min and max from water specification
+
+    >>> ranges = specification.getResultsRange()
+    >>> for range in ranges:
+    ...     range['min_operator'] = 'geq'
+    ...     range['max_operator'] = 'lt'
+    >>> specification.setResultsRange(ranges)
+
+First, get the analyses from slot 1 and sort them asc:
+
+    >>> analyses = worksheet.get_analyses_at(1)
+    >>> analyses.sort(key=lambda analysis: analysis.getKeyword(), reverse=False)
+
+Set results for analysis `Au` (min: -5, max: 5, warn_min: -5.5, warn_max: 5.5):
+
+    >>> au_analysis = analyses[0]
+    >>> au_analysis.setResult(-5)
+    >>> is_out_of_range(au_analysis)
+    (False, False)
+
+    >>> au_analysis.setResult(5)
+    >>> is_out_of_range(au_analysis)
+    (True, False)
+
+
+Check if results are out of range when right-open interval is used
+------------------------------------------------------------------
+
+Set right-open interval for min and max from water specification
+
+    >>> ranges = specification.getResultsRange()
+    >>> for range in ranges:
+    ...     range['min_operator'] = 'gt'
+    ...     range['max_operator'] = 'leq'
+    >>> specification.setResultsRange(ranges)
+
+First, get the analyses from slot 1 and sort them asc:
+
+    >>> analyses = worksheet.get_analyses_at(1)
+    >>> analyses.sort(key=lambda analysis: analysis.getKeyword(), reverse=False)
+
+Set results for analysis `Au` (min: -5, max: 5, warn_min: -5.5, warn_max: 5.5):
+
+    >>> au_analysis = analyses[0]
+    >>> au_analysis.setResult(-5)
+    >>> is_out_of_range(au_analysis)
+    (True, False)
+
+    >>> au_analysis.setResult(5)
+    >>> is_out_of_range(au_analysis)
+    (False, False)
+
+
+Check if formatted interval is rendered properly
+------------------------------------------------
+
+Set closed interval for min and max from water specification
+
+    >>> ranges = specification.getResultsRange()
+    >>> for range in ranges:
+    ...     range['min_operator'] = 'geq'
+    ...     range['max_operator'] = 'leq'
+    >>> specification.setResultsRange(ranges)
+
+Get the result range for `Au` (min: -5, max: 5)
+
+    >>> rr = specification.getResultsRange()
+    >>> res_range = filter(lambda item: item.get('keyword') == 'Au', rr)[0]
+    >>> get_formatted_interval(res_range)
+    '[-5,5]'
+
+And test after changing decimal mark
+
+    >>> bikasetup.setResultsDecimalMark(',')
+    >>> get_formatted_interval(res_range)
+    '[-5;5]'
```

Comment:
Ah, I see. You use the different results mark. Anyhow, I would suggest to stick to `;`